### PR TITLE
WIP: feat(TemplateLibrary): alternate way to add Content Control - #55

### LINKED ISSUE #55: Alternate way to restricting editing content in content controls
An alternate way to restrict editing of contents of the content control.
# Discussion 🗣

Instead of the chunk of code [here (111-119)](https://github.com/accordproject/cicero-word-add-in/blob/master/src/components/TemplateLibrary/index.js#L111), we can probably use this [API](https://docs.microsoft.com/en-us/javascript/api/word/word.contentcontrol?view=word-js-preview#cannotEdit) to do the same. Not sure how will this integrate but we can try if using this API works out.

### PR DESCRIPTION
**Pull request by Aniruddha-Shriwant**
Signed-off-by: Aniruddha Shriwant <aniruddhashriwant@gmail.com>

# Closes #55<CORRESPONDING ISSUE NUMBER>

### Changes

- The [`insertOoxml`](https://docs.microsoft.com/en-us/javascript/api/word/word.body?view=word-js-preview#insertOoxml_ooxml__insertLocation_) returns a [`Word.Range`](https://docs.microsoft.com/en-us/javascript/api/word/word.range?view=word-js-preview) 
https://github.com/accordproject/cicero-word-add-in/blob/27de6691e32d69bfc6ad1cda24145efbbd2341cc/src/components/TemplateLibrary/index.js#L140
So using Word.Range's method called [`insertContentControl()`](https://docs.microsoft.com/en-us/javascript/api/word/word.range?view=word-js-preview#insertContentControl__) an alternate way to adding content controls is added<ONE>
- As we have to restrict the user to edit the clause hence,  [`cannotEdit`](https://docs.microsoft.com/en-us/javascript/api/word/word.contentcontrol?view=word-js-preview#cannotEdit) API is used for that purpose<TWO>
- Some linting errors are fixed, I may revert them if they are not necessary for this PR

### Screenshots or Video

image
As you can see the content controls are added successfully

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [x] Merging to `master` from `fork:branchname`
- [ ] Manual accessibility test performed
    - [ ] Keyboard-only access, including forms
    - [ ] Contrast at least WCAG Level A
    - [ ] Appropriate labels, alt text, and instructions

### CODE REVIEW

File: src/components/TemplateLibrary/index.js [hunk 1]
```diff
@@ -215,7 +207,7 @@ const LibraryComponent = () => {
                   }
                 }
               }
-              if(!identifierExists){
+              if (!identifierExists) {
```
Comment (Aniruddha-Shriwant) [inline src/components/TemplateLibrary/index.js:210]: There were some linting errors that were auto fixed by calling `npm run lint:fix` I may revert these changes as they are not applicable for this PR

File: src/components/TemplateLibrary/index.js [hunk 2]
```diff
@@ -137,7 +126,10 @@ const LibraryComponent = () => {
       </pkg:part>
       ${spec}
       </pkg:package>`;
-      context.document.body.insertOoxml(ooxml, Word.InsertLocation.end);
+      const range = context.document.body.insertOoxml(ooxml, Word.InsertLocation.end);
+      const control = range.insertContentControl();
```
Comment (algomaster99) [inline src/components/TemplateLibrary/index.js]: Change the variable name to `contentControl`.

**Review by algomaster99 — APPROVED** [review]
A minor change. Rest looks great to me :)

**Comment by algomaster99** [comment]
@Aniruddha-Shriwant Forgot to suggest one more thing. We can also make these changes for the `insertVariable` method in `utils/CiceroMarkToOOXML`.

File: src/components/TemplateLibrary/index.js (hunk 2 above)
Comment (Aniruddha-Shriwant) [reply src/components/TemplateLibrary/index.js]: Okay Sure :)

**Comment by Aniruddha-Shriwant** [comment]
> @Aniruddha-Shriwant Forgot to suggest one more thing. We can also make these changes for the `insertVariable` method in `utils/CiceroMarkToOOXML`.

@algomaster99 I'm not sure how to integrate these similar changes in `utils/CiceroMarkToOOXML.js`  for `insertVariable` but I will give it a try :)

**Comment by Aniruddha-Shriwant** [comment]
I apologize for re-requesting the review 😄 I just clicked it by mistake :)

**Comment by algomaster99** [comment]
@Aniruddha-Shriwant are you still working on this?

**Comment by Aniruddha-Shriwant** [comment]
> @Aniruddha-Shriwant are you still working on this?

@algomaster99 I apologize for the delay I was getting some errors in debugging also I was a little confused about the implementation part

**Comment by algomaster99** [comment]
No problem. I was just taking an update :)

**Comment by Aniruddha-Shriwant** [comment]
> No problem. I was just taking an update :)

@algomaster99 I will work on this soon and maybe I would need your little help on this :)

**Comment by Aniruddha-Shriwant** [comment]
@algomaster99, I had spent some time on this but I was not able to add the `cannotEdit` API to `utils/CiceroMarkToOOXML.js` for `insertVariable` as I was getting some problems in getting all the variables present in the document in the `Range` class format because without it I won't be able to add contentControls on them, also the current way of adding contentControl is working fine, maybe we can do this later in another PR. 
What are your thoughts on this?
